sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/V2WriteAnalysisSuite.scala

```diff
@@ -423,8 +423,8 @@ abstract class V2WriteAnalysisSuiteBase extends AnalysisTest {
     assertNotResolved(parsedPlan)
     assertAnalysisErrorCondition(
       parsedPlan,
-      expectedErrorCondition = "INCOMPATIBLE_DATA_FOR_TABLE.CANNOT_FIND_DATA",
-      expectedMessageParameters = Map("tableName" -> "`table-name`", "colName" -> "`x`")
+      expectedErrorCondition = "INCOMPATIBLE_DATA_FOR_TABLE.EXTRA_COLUMNS",
```

Comment:
This is because of `spark.sql.defaultColumn.useNullsForMissingDefaultValues` and is aligned with V1 writes.

Reply:
If `useNullsForMissingDefaultValues` is false, suppose the test works as before? Maybe we should keep original tests and add new ones?

Reply:
Sure, I am OK either way. Let me update.

Reply:
Fixed.